dm-agent/src/mqtt.rs

```diff
@@ -0,0 +1,76 @@
+use std::sync::Arc;
```

Comment:
All the traits in this module were created to ease unit testing the MQTT interactions from this dm-agent. This whole module can be moved to our `mqtt_client` library code if there's interest.